go/arrow/flight/server.go

```diff
@@ -52,14 +52,9 @@ type (
 	Empty                           = flight.Empty
 )
 
-// FlightService_ServiceDesc is the grpc.ServiceDesc for the FlightService
-// server. It should only be used for direct call of grpc.RegisterService,
-// and not introspected or modified (even as a copy).
-var FlightService_ServiceDesc = flight.FlightService_ServiceDesc
-
```

Comment:
Why is this being removed? We added it to make it easy for others to incorporate a Flight Server into an existing grpc service.

Reply:
I got the following error with this:

```text
 arrow/flight/server.go:58:40: undefined: flight.FlightService_ServiceDesc
```

`FlightService_ServiceDesc` was renamed to `_FlightService_ServiceDesc` (`_` was prepended) in generated `Flight_grpc.pb.go`: https://github.com/apache/arrow/pull/35178/files#diff-3b9deca07311cc509c9563b866266d044d8d368cc3225f8c4524172576c50f43R632

And we can't use it because it's not exported:

```text
arrow/flight/server.go:58:40: _FlightService_serviceDesc not exported by package flight
```

If this is important for us, could you regenerate `go/arrow/flight/internal/flight/{Flight.pb.go,Flight_grpc.pb.go}` with old `protoc` and push them to this branch? I don't have old `protoc`...

See also https://github.com/apache/arrow/pull/35178/files#diff-8739a8ac1525784934c2c972465c29a068f72cb631ab2e6b36f74cfb2849abaeR21 for `protoc` versions.

Reply:
Interesting, i see that the updated versions for the `Register` method actually takes a `*grpc.Server` which should likely be sufficient to add a flight service to an existing server. So I think this is fine, but we should definitely mark this as a breaking change as a result.

Reply:
OK.
I also add "**This PR includes breaking changes to public APIs.**" to the pull request description.